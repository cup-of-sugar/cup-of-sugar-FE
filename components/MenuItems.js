import React from "react";
import {
  AsyncStorage,
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
  Image
} from "react-native";
import Colors from "../constants/Colors";
import { useNavigation } from "@react-navigation/native";

export default function MenuItems({ closeMenu, action, userId }) {
  const handlePress = path => {
    const navigation = useNavigation();
    navigation.navigate(path, { action, userId });
    closeMenu();
  };

  const handleLogout = () => {
    const navigation = useNavigation();
    try {
      AsyncStorage.removeItem("action").then(() =>
        navigation.navigate("Login")
      );
    } catch (exception) {
      console.log("Error clearing storage");
    }
  };

  return (
    <View style={styles.drawerItems}>
      <TouchableOpacity onPress={() => handlePress("Home", { action, userId })}>
        <Text style={styles.drawerText}>Home</Text>
      </TouchableOpacity>
      <TouchableOpacity onPress={() => handlePress("Path")}>
        <Text style={styles.drawerText}>
          Go to {action === "borrow" ? "Lending" : "Borrowing"}
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={() => handlePress("My Items", { action, userId })}
      >
        <Text style={styles.drawerText}>
          {action === "borrow" ? "My Borrowed Items" : "Items I'm Lending"}
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={() => handlePress("Offers/Requests", { action, userId })}
      >
        <Text style={styles.drawerText}>
          {action === "borrow" ? "My Item Requests" : "Items I'm Offering"}
        </Text>
      </TouchableOpacity>
      {action === "lend" ? (
        <TouchableOpacity
          onPress={() => handlePress("LenderRequests", { action, userId })}
        >
          <Text style={styles.drawerText}>Neighorhood Requests</Text>
        </TouchableOpacity>
      ) : null}
      <TouchableOpacity
        onPress={() => handlePress("Messages", { action, userId })}
      >
        <Text style={styles.drawerText}>My Messages</Text>
      </TouchableOpacity>
      {action === "borrow" ? (
        <TouchableOpacity
          onPress={() => handlePress("Request", { action, userId })}
        >
          <Text style={styles.drawerText}>Make a Request</Text>
        </TouchableOpacity>
      ) : null}
      <TouchableOpacity onPress={() => handleLogout()}>
        <Text style={styles.drawerText}>Logout</Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  drawerItems: {
    backgroundColor: Colors.lightBlue,
    height: 490,
    width: 260
  },
  drawerText: {
    borderColor: "#fff",
    borderWidth: 1,
    color: "#fff",
    fontSize: 20,
    fontWeight: "bold",
    height: 70,
    padding: 20,
    width: "100%"
  }
});
