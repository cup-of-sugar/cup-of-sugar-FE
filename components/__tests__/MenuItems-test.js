import React from "react";
import MenuItems from "../MenuItems";

import renderer from "react-test-renderer";
import { MockedProvider } from "@apollo/react-testing";

it("should render component when action is lend", () => {
  // const props = {
  //   closeMenu: jest.fn(),
  //   action: "lend"
  // };
  // const tree = renderer
  //   .create(
  //     <MockedProvider mocks={[]}>
  //       <MenuItems {...props} />
  //     </MockedProvider>
  //   )
  //   .toJSON();
  // expect(tree).toMatchSnapshot();
});

it("should render component when action is borrow", () => {
  // const props = {
  //   closeMenu: jest.fn(),
  //   action: "borrow"
  // };
  // const tree = renderer
  //   .create(
  //     <MockedProvider mocks={[]}>
  //       <MenuItems {...props} />
  //     </MockedProvider>
  //   )
  //   .toJSON();
  // expect(tree).toMatchSnapshot();
});
