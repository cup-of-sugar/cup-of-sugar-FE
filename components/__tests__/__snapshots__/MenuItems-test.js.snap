// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component when action is borrow 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#0C94E1",
      "height": 490,
      "width": 260,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fff",
          "borderWidth": 1,
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "height": 70,
          "padding": 20,
          "width": "100%",
        }
      }
    >
      Home
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fff",
          "borderWidth": 1,
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "height": 70,
          "padding": 20,
          "width": "100%",
        }
      }
    >
      Go to 
      Lending
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fff",
          "borderWidth": 1,
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "height": 70,
          "padding": 20,
          "width": "100%",
        }
      }
    >
      My Borrowed Items
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fff",
          "borderWidth": 1,
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "height": 70,
          "padding": 20,
          "width": "100%",
        }
      }
    >
      My Item Requests
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fff",
          "borderWidth": 1,
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "height": 70,
          "padding": 20,
          "width": "100%",
        }
      }
    >
      My Messages
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fff",
          "borderWidth": 1,
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "height": 70,
          "padding": 20,
          "width": "100%",
        }
      }
    >
      Make a Request
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fff",
          "borderWidth": 1,
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "height": 70,
          "padding": 20,
          "width": "100%",
        }
      }
    >
      Logout
    </Text>
  </View>
</View>
`;

exports[`should render component when action is lend 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#0C94E1",
      "height": 490,
      "width": 260,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fff",
          "borderWidth": 1,
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "height": 70,
          "padding": 20,
          "width": "100%",
        }
      }
    >
      Home
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fff",
          "borderWidth": 1,
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "height": 70,
          "padding": 20,
          "width": "100%",
        }
      }
    >
      Go to 
      Borrowing
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fff",
          "borderWidth": 1,
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "height": 70,
          "padding": 20,
          "width": "100%",
        }
      }
    >
      Items I'm Lending
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fff",
          "borderWidth": 1,
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "height": 70,
          "padding": 20,
          "width": "100%",
        }
      }
    >
      Items I'm Offering
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fff",
          "borderWidth": 1,
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "height": 70,
          "padding": 20,
          "width": "100%",
        }
      }
    >
      Neighorhood Requests
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fff",
          "borderWidth": 1,
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "height": 70,
          "padding": 20,
          "width": "100%",
        }
      }
    >
      My Messages
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fff",
          "borderWidth": 1,
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "height": 70,
          "padding": 20,
          "width": "100%",
        }
      }
    >
      Logout
    </Text>
  </View>
</View>
`;
